debezium-server-redis/src/main/java/io/debezium/server/redis/RedisStreamChangeConsumer.java

```diff
@@ -162,24 +167,32 @@ public void handleBatch(List<ChangeEvent<Object, Object>> records,
                         LOGGER.error("Can't connect to Redis", e);
                     }
                 }
-                else if (isMemoryOk.check()) {
+                else {
                     try {
-                        LOGGER.trace("Preparing a Redis Pipeline of {} records", clonedBatch.size());
+                        LOGGER.debug("Preparing a Redis Pipeline of {} records", clonedBatch.size());
 
                         List<SimpleEntry<String, Map<String, String>>> recordsMap = new ArrayList<>(clonedBatch.size());
                         for (ChangeEvent<Object, Object> record : clonedBatch) {
                             String destination = streamNameMapper.map(record.destination());
                             Map<String, String> recordMap = recordMapFunction.apply(record);
                             recordsMap.add(new SimpleEntry<>(destination, recordMap));
                         }
+                        if (!redisMemoryThreshold.checkMemory(getObjectSize(recordsMap.get(0)), recordsMap.size(),
```

Comment:
Should guard against empty batch, would throw an excaption in this case.

Reply:
Added check for empty batch